urlc98c3b97d2_Trimax_venta_tgz-pJ8-com_gesoftware_venta_structures_map_MapTestJ8/src/com/gesoftware/venta/utility/PropertiesUtility.java

```diff
@@ -0,0 +1,88 @@
+package com.gesoftware.venta.utility;
+
+import com.gesoftware.venta.logging.LoggingUtility;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Properties;
+
+/**
+ * PropertiesUtility class definition
+ */
+public final class PropertiesUtility {
+    /* Properties loaded from configuration file */
+    private final Properties properties = new Properties();
+
+    /* *
+     * METHOD: Class constructor
+     *  PARAM: [IN] configurationFile - configuration file path
+     * AUTHOR: Eliseev Dmitry
+     * */
+    public PropertiesUtility(final String configurationFile) {
+        try {
+            properties.load(new FileInputStream(configurationFile));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource created by the following constructor: *FileInputStream*. Execution paths that do not contain closure statements were detected. Ensure the object returned by *FileInputStream()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).